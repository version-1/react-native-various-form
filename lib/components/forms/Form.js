import React, { Component } from 'react'
import { ScrollView, Alert } from 'react-native'
import ValidatorCollector from '../../validators/ValidatorCollector'
import Validator from './Validator'

export default class extends Component {
  constructor(props) {
    super(props)
    // chech validator type by initialize Validator
    this.validator
    this.validationResult = {}
  }

  validate(targets) {
    this.validationResult = this.validator.validate()
  }

  get collectValidators() {
    const { children, keyIndex, valueIndex } = this.props
    const collector = new ValidatorCollector(children, keyIndex, valueIndex)
    return collector.collect()
  }

  get validator() {
    return new Validator({
      ...this.collectValidators,
      ...this.props.validators
    })
  }

  onSubmit() {
    this.validate({ ...this.collectValidators, ...this.validators })
    this.props.onSubmit(this.validationResult)
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.trigger) {
      this.onSubmit()
    }
  }

  render() {
    const { children, style } = this.props
    return <ScrollView style={{ ...style }}>{this.props.children}</ScrollView>
  }
}
