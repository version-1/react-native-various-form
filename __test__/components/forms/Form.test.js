import Form from '../../../lib/components/forms/Form'
import React, { Component } from 'react'
import { View, TextInput, Button } from 'react-native'
import renderer from 'react-test-renderer'

describe('Form', () => {
  describe('normal case', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Form>
            <View>
              <TextInput name="email" value="" emali required />
            </View>
            <View>
              <TextInput name="password" value="" />
            </View>
            <Button title="submit" onPress={() => {}} />
          </Form>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
