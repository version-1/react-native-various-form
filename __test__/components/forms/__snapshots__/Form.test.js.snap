// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form normal case renders correctly 1`] = `
<RCTScrollView
  style={Object {}}
>
  <View>
    <View>
      <TextInput
        allowFontScaling={true}
        emali={true}
        name="email"
        required={true}
        value=""
      />
    </View>
    <View>
      <TextInput
        allowFontScaling={true}
        name="password"
        value=""
      />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          submit
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
